sdk/python/kfp/compiler/compiler.py

```diff
@@ -332,6 +332,36 @@ def _build_conventional_artifact(name):
     if op.num_retries:
       template['retryStrategy'] = {'limit': op.num_retries}
 
+    if op.sidecars:
+      def _sidecar_to_template(tsidecar):
+          index, sidecar = tsidecar
+          sidecar_template = {
+            'name': f'{op.name}-{sidecar.name}-{index}',
```

Comment:
Not sure this format is supported in Python 3.5

Reply:
replaced `f-string` with `string format` for py 3.5.